src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -237,9 +238,19 @@ private void esbRestApiCall(HttpServletRequest request,
 
 	private void apiConnectionLogic(HttpServletRequest request, HttpServletResponse response)
 			throws IOException {
-		String queryString = request.getQueryString();
 		String pathInfo = request.getPathInfo();
 		PrintWriter out = response.getWriter();
+		if(pathInfo.equalsIgnoreCase(MPATHWAYS_PATH_INFO)){
+			testMpathwaysCall(out);
+		}
+		else{
+			getCanvasResponse(request, response, out, pathInfo);
+		}
+	}
+
+	private void getCanvasResponse(HttpServletRequest request,
+			HttpServletResponse response, PrintWriter out, String pathInfo) throws IOException {
```

Comment:
Is there a reason that the pathInfo value passed in would be different from that already in the request? If not it's better to get it from the request rather than duplicate the information.